packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -353,7 +353,9 @@ export class FluidDataStoreRuntime
 	public async request(request: IRequest): Promise<IResponse> {
 		try {
 			const parser = RequestParser.create(request);
-			const id = parser.pathParts[0];
+			// TODO why are we non null asserting here?
```

Comment:
Another case where the assertion changes the type of id from string|undefined to string. Is that safe?

Reply:
Probably don't want to non-null-assert here, it's not guaranteed for pathParts to have length > 0

Reply:
What should I do in this case?

Reply:
Below where it gets `this.contexts.get(id)`, if id is undefined it won't find a context and will then return a 404 response (which is good - luckily the improper handling of undefined here still results in the behavior we want).  There are several ways you could choose to restructure to maintain that behavior while explicitly handling the undefined that could be fine.  I'd default to wrapping that context check in a conditional which checks if id is not undefined.